### PR TITLE
APIv3 Version list slug filter added

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/readthedocs/readthedocs.org/issues/7363

### CODE REVIEW

File: docs/api/v3.rst [hunk 1]
```diff
@@ -412,6 +412,10 @@ Versions listing
 
     :query boolean active: return only active versions
     :query boolean built: return only built versions
+    :query string privacy_level: return versions with specific privacy level
+    :query string slug: return versions with matching slug
+    :query boolean uploaded: return only uploaded versions
```
Comment (stsewd) [inline docs/api/v3.rst:417]: Uploaded shouldn't be here, or at least it shouldn't be documented. This is a very old feature we don't expose anymore. And I'm not really sure we have versions marked as uploaded.

File: docs/api/v3.rst [hunk 2]
```diff
@@ -412,6 +412,10 @@ Versions listing
 
     :query boolean active: return only active versions
     :query boolean built: return only built versions
+    :query string privacy_level: return versions with specific privacy level
```
Comment (stsewd) [inline docs/api/v3.rst:415]: we should document the options here: `private` or `public`

File: readthedocs/api/v3/filters.py [hunk 3]
```diff
@@ -25,6 +27,7 @@ class Meta:
             'active',
             'built',
             'uploaded',
+            'slug',
```
Comment (stsewd) [inline readthedocs/api/v3/filters.py:30]: I think type should be a good filter too.

File: docs/api/v3.rst (hunk 1 above)
Comment (saadmk11) [reply docs/api/v3.rst]: Okay, sure! I did not know that. Removed it from documentation but not sure if I should remove it from filters as it was already there.

Comment (stsewd) [reply docs/api/v3.rst]: I think it was added in the filters to make use of it internally only

File: docs/api/v3.rst [hunk 4]
```diff
@@ -412,6 +412,10 @@ Versions listing
 
     :query boolean active: return only active versions
     :query boolean built: return only built versions
+    :query string privacy_level: return versions with specific privacy level (``public`` or ``private``)
+    :query string slug: return versions with matching slug
+    :query string type: return versions with specific type (``branch`` or ``tag``)
+    :query string verbose_name: return versions with matching version name
```
Comment (agjohnson) [inline docs/api/v3.rst:418]: :100: on doc updates

**Review by agjohnson — APPROVED** [review]
Looks great, exactly what i need!

**Pull request merged**